File: src/CompHub.ReactApp/src/types/order.ts
export type OrderStatus = 'Pending' | 'Processing' | 'Shipped' | 'Delivered' | 'Cancelled';

export interface OrderListItem {
    id: number;
    createdAt: string;
    status: OrderStatus;
    totalAmount: number;
    itemCount: number;
}

export interface OrderItemDto {
    productId: number;
    productName: string;
    productImageUrl?: string;
    quantity: number;
    unitPrice: number;
    subtotal: number;
}

export interface OrderDetail {
    id: number;
    createdAt: string;
    status: OrderStatus;
    totalAmount: number;
    shippingAddress: string;
    shippingCity: string;
    items: OrderItemDto[];
}

export interface CreateOrderRequest {
    shippingAddress: string;
    shippingCity: string;
    items: { productId: number; quantity: number }[];
}

File: src/CompHub.WebApi/Domain/Entities/Order.cs
using System.ComponentModel.DataAnnotations;

namespace CompHub.WebApi.Domain.Entities
{
    public enum OrderStatus
    {
        Pending = 0,
        Processing = 1,
        Shipped = 2,
        Delivered = 3,
        Cancelled = 4
    }

    public class Order
    {
        public int Id { get; set; }
        public DateTime CreatedAt { get; set; }
        public OrderStatus Status { get; set; } = OrderStatus.Pending;
        public decimal TotalAmount { get; set; }

        // Snapshot of shipping address at the time of order
        public string ShippingAddress { get; set; } = string.Empty;
        public string ShippingCity { get; set; } = string.Empty;

        public int UserId { get; set; }
        public User User { get; set; } = null!;

        public ICollection<OrderItem> Items { get; set; } = new List<OrderItem>();
    }
}
